### PR TITLE
feat(options): prompt to enable experimental affinity(#1051, #1267)

### PR DESCRIPTION
**Pull request by prabin-lc**
Check if the experimental.affinity settings in User or Workspace settings is assigned for this extension. If not, set the affinity value to the default(1).

Then, prompt to restart the extension host which is required for the changes to take effect. This will restart all extensions.

### CODE REVIEW

**Comment by prabin-lc** [comment]
regarding #1051, #1267

**Review by theol0403 — APPROVED** [review]
Well done! The code looks good, I'll merge once I get a chance to test.

File: src/main_controller.ts [hunk 1]
```diff
@@ -368,4 +370,67 @@ export class MainController implements vscode.Disposable {
...
+                    },
+                    (error) => {
+                        this.logger.error(`Error while setting experimental affinity. ${error}`);
+                    },
+                );
+        };
+
+        vscode.window
+            .showWarningMessage(
+                "No affinity assigned to vscode-neovim. It is recommended to assign affinity for major performance improvements. Would you like to set default affinity?",
```
Comment (justinmk) [inline src/main_controller.ts:423]: vscode messages support URLs. Would be very nice to link to the relevant documentation here.
```suggestion
                "No [affinity](https://github.com/vscode-neovim/vscode-neovim/issues/1051) assigned to vscode-neovim. It is recommended to assign affinity for major performance improvements. Would you like to set default affinity?",
```

Comment (prabin-lc) [reply src/main_controller.ts]: I have added a paragraph on affinity on the docs and added a link to it on the prompt. The discussions on #1051 don't quiet explain the feature.

File: README.md [hunk 2]
```diff
@@ -63,13 +63,21 @@ mode and editor commands, making the best use of both editors.
...
 
-```json
-"extensions.experimental.affinity": {
-    "asvetliakov.vscode-neovim": 1
-},
-```
+    -   Settings > Features > Extensions > Experimental Affinity.
+
+            Assign value 1 for entry `asvetliakov.vscode-neovim`
+            ![Alt text](images/experimental_affinity.png)
```
Comment (justinmk) [inline README.md:71]: Don't think an image is worth adding to the repo. Eventually vscode will get a proper doc for this, then we'll link to it. This is just temporary.

Comment (prabin-lc) [reply README.md]: Makes sense. Removed the image.

File: images/experimental_affinity.png
Comment (justinmk) [inline images/experimental_affinity.png:1]: the image is still in the PR

Comment (prabin-lc) [reply images/experimental_affinity.png:1]: Oops. Sorry about that. Removed the image as well.

**Pull request merged**